### PR TITLE
Enable new inline rename by default

### PR DESCRIPTION
**Pull request by ryzngard**

### CODE REVIEW

**Comment by JoeRobich** [comment]
Should there also be a VB.NET setting for this?

**Comment by ryzngard** [comment]
> Should there also be a VB.NET setting for this?

This is where we get a little tricky. I can duplicate in VB, but it only applies globally not per language. I guess we could do per language...

Unfortunately we don't have a ".NET" node to put stuff like this in :(

File: src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml [hunk 1]
```diff
@@ -83,6 +89,16 @@
                 <StackPanel>
                     <CheckBox x:Name="Rename_asynchronously_exerimental"
                               Content="{x:Static local:AdvancedOptionPageStrings.Option_Rename_asynchronously_experimental}" />
+
+                    <Label x:Name="Rename_UI_setting_label" Content="{x:Static local:AdvancedOptionPageStrings.Where_should_the_rename_ui_be_shown}" />
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml:93]: a bit odd that `ui` is lowercased.

Comment (CyrusNajmabadi) [reply src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml:93]: maybe just `Rename textbox location: inline/in-top-right`?

File: src/VisualStudio/Core/Def/ServicesVSResources.resx [hunk 2]
```diff
@@ -2004,4 +2004,13 @@ Additional information: {1}</value>
   <data name="Document_Outline" xml:space="preserve">
     <value>Document Outline</value>
   </data>
+  <data name="Show_ui_as_dashboard_in_top_right" xml:space="preserve">
+    <value>Show UI as a dashboard in top right</value>
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Def/ServicesVSResources.resx:2008]: ah, you are using UI in the actual user string.  then can you update hte loc key to match that?  (note this is a benefit of keeping both in sync. it means at teh code point it's clear what the user will see.

File: src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml (hunk 1 above)
Comment (ryzngard) [reply src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml]: That's not completely accurate though. It's a bit more nuanced than that. At most, inline rename options or dashboard them?

**Review by Youssef1313 — COMMENTED** [review]
Should this be added to `AutomationObject`?

I'm concerned by this being added both to C#-specific and VB-specific options page but have a single value for both (ie, if user modified it in one language, it gets automatically the same value in the other).

Are there other similar options that are single-valued for both languages and exposed in Tools → Options?

Ideally there should be some sort of a "common" page. @mavasani What do you think?

File: src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml [hunk 3]
```diff
@@ -83,6 +89,16 @@
                 <StackPanel>
                     <CheckBox x:Name="Rename_asynchronously_exerimental"
                               Content="{x:Static local:AdvancedOptionPageStrings.Option_Rename_asynchronously_experimental}" />
+
+                    <Label x:Name="Rename_UI_setting_label" Content="{x:Static local:AdvancedOptionPageStrings.Where_should_the_rename_UI_be_shown}" />
+                    <ComboBox x:Name="Rename_UI_setting">
```
Comment (Youssef1313) [inline src/VisualStudio/CSharp/Impl/Options/AdvancedOptionPageControl.xaml:94]: Why not a checkbox?

**Comment by davidwengier** [comment]
> Are there other similar options that are single-valued for both languages and exposed in Tools → Options?

Allow navigation to decompiled sources is for both languages, and only appears on the C# page, presumably because there is nowhere else to put it.

**Comment by ryzngard** [comment]
> Should this be added to `AutomationObject`?
> 
> I'm concerned by this being added both to C#-specific and VB-specific options page but have a single value for both (ie, if user modified it in one language, it gets automatically the same value in the other).
> 
> Are there other similar options that are single-valued for both languages and exposed in Tools → Options?
> 
> Ideally there should be some sort of a "common" page. @mavasani What do you think?

I haven't kept up with `AutomationObject`. Is there  a doc I can look at to what needs to go there and why? Or just a short description

**Comment by Youssef1313** [comment]
@ryzngard I don't know of docs about it, but basically you can observe the impact by:

1. Tools → Import and Export Settings
2. Follow with the dialog to export settings into a file
3. Change the value of the option of interest to test
4. Import the file back

When importing the file back, the old value should be seen.

**Comment by JoeRobich** [comment]
> Are there other similar options that are single-valued for both languages and exposed in Tools → Options?

Editor Color Scheme falls in this bucket. Both pages bind to the same option (not perlanguage) and the OptionStore class ensures that changes are synced.

**Pull request merged**